src/lib/xocl-subdev.c

```diff
@@ -268,6 +269,10 @@ xocl_subdev_create(struct device *parent, enum xocl_subdev_id id,
 	if (sysfs_create_group(&DEV(pdev)->kobj, &xocl_subdev_attrgroup))
 		xocl_err(pdev, "failed to create sysfs group");
 
+	drvdata = DEV_DRVDATA(pdev);
+	/* All done, ready to handle req thru cdev. */
+	if (drvdata->xsd_dev_name)
+		(void) xocl_devnode_create(pdev, drvdata->xsd_dev_name, NULL);
```

Comment:
How do we want to support an instance of a subdev to provide a instance specific devnode name? @houlz0507 has a use case for this.

Reply:
I would recommend to expose xocl_devnode_create() to leaf driver. This give leaf driver to generate a meaningful device node name.
   1. it could be very useful for the case that the same subdevice has multiple instances. Using meaningful device node name, the application, tools, scripts and driver code can be simplified. (userland code does not have to loop all the device node to determine which one it needs to open. driver does not have to implement ioctl just to identify the instance)
   2. Using a meaningful device node name is also very helpful for debugging. For example, if customer give a log which said open /dev/xfpga/cu12 failed, then we can immediately know that cu12 has problem if 12 is meaningful.
   3. Why bother to call  xocl_devnode_create() in leaf driver? It does not sound hurt anything to me.